clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -910,8 +874,11 @@ private void sendProduceRequest(long now, int destination, short acks, int timeo
         RequestCompletionHandler callback = response -> handleProduceResponse(response, recordsByPartition, topicNames, time.milliseconds());
 
         String nodeId = Integer.toString(destination);
+        
+        long deliveryTimeoutMs = accumulator.getDeliveryTimeoutMs() - (now - earliestCreatedMs);
```

Comment:
Hmm, this is a bit more complicated than I thought since there are multiple batches with different creation time in a request. For now, it's probably better to just use requestTimeoutMs. This could extend the delivery timeout by request timeout in the rare cases. We can file a followup jira to see if there are better solutions in the future.